authors/apps/authentication/tests/test_views.py

```diff
@@ -0,0 +1,70 @@
+from rest_framework.views import status
+
+from .test_base import BaseViewTest
+class RegistrationLoginViewTest(BaseViewTest):
+
+    def test_register_new_user_correct_details(self):
+        """Test api  can create user"""
+        response = self.register_user("testuser", "test.sims@andela.com", "testman101")
+
+        self.assertEqual(response.data['email'], "test.sims@andela.com")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_register_with_weak_password(self):
+        """Test api  can't create user if password is invalid"""
+
+        response = self.register_user("testuser", "test.sims@andela.com", "test")
+
+        self.assertEqual(response.data['errors']['password'], ["Ensure this field has at least 8 characters."])
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_login_with_valid_credentials(self):
+        """Test api can login user if user password and email match"""
+
+        response = self.log_in_user("simulations.sims@andela.com", "simulman101")
+
+        self.assertEqual(response.data['username'], "simuser")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_login_with_incorrect_credentials(self):
+        """Test api can't login user if user password and email don't match"""
+        
+        response = self.log_in_user("fahad.sims@andela.com", "manny101")
+
+        self.assertEqual(response.data['errors']['error'], ["A user with this email and password was not found."])
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+class UserRetrieveUpdateTest(BaseViewTest):
+
+    def test_get_user_credentials_when_authenticated(self):
+        """Test api can get user if user is authenticated"""
+        
+        # authenticate user
+        self.client.force_authenticate(user=self.user)
+
+        # get user data
+        url = '/api/user/'
+        response = self.client.get(url)
+
+        self.assertEqual(response.data['username'], "jealous")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_edit_user_credentials(self):
+        """Test api can edit user data if user is authenticated"""
+        
+        # authenticate user
+        self.client.force_authenticate(user=self.user)
+
+        user_update = {
```

Comment:
@fahadmak you should create a test fixtures file and put this data there and import it to use here.